server/src/main/java/org/opensearch/cluster/node/DiscoveryNodeRole.java

```diff
@@ -273,6 +274,10 @@ public void validateRole(List<DiscoveryNodeRole> roles) {
             }
         }
 
+        @Override
+        public boolean isEnabledByDefault(final Settings settings) {
+            return !Booleans.isBoolean(settings.get("node.master"));
```

Comment:
Nitpick, the the prevailing style is `Booleans.isBoolean(settings.get("node.master")) == false`

Reply:
Thanks for reviewing @andrross! Addressed the comment and re-based my changes against main to resolve conflicts in *CHANGELOG.md*